services/src/pro/datasets/postgres.rs

```diff
@@ -68,7 +79,7 @@ where
             pos += 1;
             (format!("AND d.tags @> ${pos}::text[]"), filter_tags.clone())
         } else {
-            (String::new(), vec![])
+            ("AND NOT d.tags @> '{deleted}'::text[]".to_string(), vec![])
```

Comment:
if the tags contain "deleted" it still would filter out the deleted datasets right?

Reply:
The intention was:
If tags contains deleted, then it will list deleted datasets. 
If the tags contain anything else, but not the deleted tag, then it will not list deleted datasets (because AND d.tags does not search for it.)
If the tags do not contain anything (case of the marked line), then do not list deleted datasets, hence the not contains. 

I hope it this covers it. The discussed alternative (negative tag filter) was seemingly not desired, so I did not implement it.